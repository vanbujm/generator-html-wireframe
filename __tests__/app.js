/* eslint-disable no-undef */
import 'babel-polyfill';
import path from 'path';
import assert from 'yeoman-assert';
import helpers from 'yeoman-test';
import fs from 'fs';

let yeomanContext;
const testName = 'testName';

describe('generator-html-wireframe:app', () => {
  beforeAll(async() => {
    yeomanContext = await helpers.run(path.join(__dirname, '../src/app/index.js'))
      .withPrompts({appName: testName, nav: 'Fixed'});
  });

  test('should write html and css files', () => {
    assert.file(['html/' + testName + '-wireframe.html']);
    assert.file(['css/' + testName + '-style.css']);
  });

  test('should generate test wireframe', () => {
    const file = fs.readFileSync('html/' + testName + '-wireframe.html', 'utf8');
    expect(file).toMatchSnapshot();
  });

});
