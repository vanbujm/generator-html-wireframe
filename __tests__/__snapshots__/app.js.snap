// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator-html-wireframe:app should generate test wireframe 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <title>testName Wireframe</title>
  <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"../css/testName-style.css\\">
</head>
<body>

<nav class=\\"nav\\">
  <ul class=\\"left\\">
    <li class=\\"brand\\">
      <a href=\\"\\">brand</a>
    </li>
    <li>
      <a href=\\"\\">item 1</a>
    </li>
    <li>
      <a href=\\"\\">item 2</a>
    </li>
    <li>
      <a href=\\"\\">item 3</a>
    </li>
  </ul>
  <ul class=\\"right\\">
    <li>
      <a href=\\"\\">right item 4</a>
    </li>
    <li class=\\"hover-container\\">
      <a href=\\"\\">hover list item 5</a>
      <ul class=\\"hover-menu\\">
        <li>hover item 1</li>
        <li>hover item 2</li>
        <li>hover item 3</li>
        <li>hover item 4</li>
      </ul>
    </li>
  </ul>
</nav>

<div class=\\"content\\">
  <h1>This page was generated by html-wireframe generator!</h1>
  <h2> Your answers:</h2>
  <h3>
    <ul>
      <li>appName: testName</li>
      <li>nav: Fixed</li>
    </ul>
  </h3>
  <hr>
  <p>
    Mongrel sunnies veg out knock bradman lurk. Frog in a sock milk bar pint grog cranky. Back of bourke deadset struth
    rollie freckle. Icy pole what's crackin' bottle-o his blood's worth bottling bitzer rack off lippy. Gone walkabout
    rage on what's crackin' struth sleepout freo porky dinky-di rack off kindie. Crook cranky mozzie flanny budgie
    smugglers cark it as cross as a rort flat out like a. Joey turps down under gutful of heaps larrikin shazza got us
    some. Bradman ya flat out like a whinge bail out cockie dob coldie. Bities pash waggin' school cut snake chokkie
    with vee dub wuss.
  </p>
  <p>
    Good oil rock up muster when boil-over mickey mouse mate. As cross as a thingo oldies relo bog standard troppo
    aerial pingpong. Cactus mate thongs stonkered billy buckley's chance ocker good onya give it a burl. Rack off
    jillaroo coathanger blowie billabong middy dero. Built like a grundies butcher veg out it'll be as stands out like.
    Ocker reckon dole bludger barbie servo trackies wobbly dole bludger galah.
  </p>
  <p>
    Fairy floss garbo moolah captain cook aussie rules footy. Jug brumby rollie longneck. Flick muster pint true blue
    paddock spag bol cut snake. As cross as a built like a avos prezzy bizzo gutta shazza stonkered whinge. Brumby
    vinnie's drongo budgie smugglers cane toad you little ripper as cross as a no dramas you little ripper. Rack off
    coldie gutta veg out fair go pash ya sook galah ute. Kelpie divvy van squizz mate no dramas brizzie bull bar ten
    clicks away metho. Outback as stands out like barrack dag sleepout chuck a yewy wobbly too right. Stubby bushranger
    wobbly gutta to stubby. Servo freckle stubby holder ugg boots.
  </p>
  <p>
    How stands out like a ridgy-didge bog standard cark it blowie roo hit the turps. Mate cockie quid too right deadset
    cleanskin ugg boots cab sav stickybeak crook. Shazza ironman ciggies piker uluru hit the turps boozer spit the dummy
    bogan. Troppo cranky bottlo chrissie as dry as a sheila grab us a pretty spiffy rotten boogie board we're going
    hottie. Blow in the bag wobbly stoked bloke fair go no-hoper cut lunch commando rapt flanny. Rock up boogie board
    spit the dummy bail up turps throw-down cracker. Slaps mokkies lurk too right bloody gutta.
  </p>
  <p>
    Slabs thingo damper cooee ocker longneck bludger grouse. No dramas struth barbie captain cook piece of piss rollie
    fly wire. Spewin' compo rollie grouse boil-over ankle biter middy. Quid going off arvo jug bludger. Boil-over budgie
    smugglers show pony too right swagger strewth come a fly wire cockie dunny bushranger chokkie. Ambo banana bender
    cleanskin thongs wobbly gone walkabout cane toad dropkick. Milk bar dinky-di sheila rack off brekkie. Shonky flamin
    mates bities damper slacker fairy floss coathanger bonzer grab us a deadset. Stubby holder brisvegas clacker dog's
    balls freo not my bowl of rice how cleanskin. Grab us a flanny divvy van bradman stubby wuss bounce rip snorter
    no-hoper.
  </p>
  <p>
    Cark it cook vinnie's rip snorter where dero cockie. Boardies ratbag dag trent from punchy. Chuck a yewy paddock no
    worries ute. Offsider mad as a slab troppo cubby house fossicker mad as a ocker. Beauty bitzer bottle-o lizard
    drinking thingo tucker. Sickie digger fairy floss grab us a brizzie grab us a cook. Sleepout hottie dunny ratbag
    cactus mate. Ankle biter maccas booze bus gobful avos dinky-di.
  </p>

  <p>Tucker butcher bonza chrissie bonza turps. Daks dunny cook barbie bottlo cut lunch commando how relo give it a
    burl. Truckie bottle-o mokkies sheila booze chuck a sickie garbo top end squizz grundies. Larrikin mokkies fair go
    prezzy slacker banana bender hooroo. Jumbuck spewin' dinky-di wuss smokes. Rego ropeable jumbuck yobbo. Bundy relo
    ugg bogan bloody oath as dry as a. Cab sav dero bunyip coathanger figjam christmas. Lizard drinking avos bities cark
    it pash dill.
  </p>

  <p>Going off back of bourke hottie rip snorter boogie board apples booze bus ropeable spit the dummy holden. Franger
    veg out dinky-di stoked struth icy pole. Booze thongs bizzo tinny sickie. Sheila holy dooley crook trent from punchy
    bazza throw-down going off. He hasn't got a your shout sleepout roadie damper cooee kelpie fly wire. Stands out like
    a and relo uluru cook throw-down gutful of slabs sickie coathanger cream.
  </p>

  <p>Vb your shout lippy rapt blow in the bag stands out like a rubbish bog standard ace larrikin. Stubby bunyip barbie
    schooner dunny dob prezzy. Barbie booze bush oyster shag on a rock lets get some cut snake swagger ute sunnies
    dinky-di avos schooner. Pot thongs ya rage on gyno bludger gutful of two up bazza stands out like a. Billy quid
    metho rego g'day trackie dacks coldie sheila ute. Sickie no-hoper sanger pav cut snake shazza got us some. Sickie as
    dry as a moolah good oil bloke bonza cab sav cream servo. Ropeable gone walkabout brickie grundies wuss g'day he
    hasn't got a. Bloody whinge good onya rage on spewin' dinky-di brekkie shag on a rock vb. Wobbly hit the turps divvy
    van schooner chook. Rubbish mate crook rollie bikie bush bash maccas coldie compo. Knock holden doovalacky gyno
    vinnie's gobful mozzie. Dunny bunyip rubbish larrikin what's crackin' bundy oi you little ripper sickie sleepout
    stoked.
  </p>

  <p>As cunning as a strides garbo lippy christmas roo ace piker. Good oil middy corker ropeable porky. Jug dole bludger
    deadset postie bog standard slaps bounce we're going. To cark it porky show pony dead horse avos cranky ratbag
    holden. Tinny rotten stands out like a as busy as a muster. Your shout sickie clucky damper top end metho jug.
    Rollie bail out ironman cobber fisho knock. Shazza got us some pint no dramas pav spit the dummy dead horse rage on
    flat out like a trent from punchy. Rubbish slacker spag bol bail up longneck lurk daks greenie.
  </p>

  <p>Doovalacky you little ripper ripper chunder chuck a spaz. Sheila not my bowl of rice nipper bail up trent from
    punchy. Brizzie buckley's chance brass razoo coathanger mate galah dero. Dole bludger bail out aussie salute spit
    the dummy gyno big smoke. Hooroo boil-over clacker dinky-di bushranger piece of piss pokies chunder bonza. Shazza
    got us some bloke dunny rat ya pozzy bonzer trackies deadset ocker sanger. Down under swag strewth spag bol ute
    cockie ugg boots oi piker barbie.
  </p>

  <p>Relo flanny icy pole your shout rollie aussie rules footy plonk bush oyster tucker you little ripper. When bogan
    yabber fly wire fruit loop dole bludger moolah ocker. How blow in the bag barbie arvo chunder. Parma ute mate slabs
    oi stonkered boogie board pozzy hooroo knock. Divvy van wuss rapt drongo blowie spag bol too right yabber ocker roo
    bar. She'll be right ropeable ute trackie dacks piece of piss sunnies chuck a yewy cook barbie. Plonk blue battler
    chokkie throw-down. Boardies daks bundy dunny rock up mokkies back of bourke sanger tinny cook. Kindie what's
    crackin' sheila chuck a sickie shonky.
  </p>

</div>

</body>
</html>
"
`;
